server/src/main/java/org/candlepin/resource/OwnerProductResource.java

```diff
@@ -458,19 +463,23 @@ public void deleteProduct(
     @Produces(MediaType.APPLICATION_JSON)
     @Consumes(MediaType.WILDCARD)
     @Transactional
-    public JobDetail refreshPoolsForProduct(
+    public AsyncJobStatusDTO refreshPoolsForProduct(
         @PathParam("owner_key") String ownerKey,
         @PathParam("product_id") String productId,
-        @QueryParam("lazy_regen") @DefaultValue("true") Boolean lazyRegen) {
+        @QueryParam("lazy_regen") @DefaultValue("true") boolean lazyRegen) throws JobException {
 
         if (config.getBoolean(ConfigProperties.STANDALONE)) {
             log.warn("Ignoring refresh pools request due to standalone config.");
             return null;
         }
 
-        Owner owner = this.getOwnerByKey(ownerKey);
-        Product product = this.fetchProduct(owner, productId);
+        final Owner owner = this.getOwnerByKey(ownerKey);
+        final Product product = this.fetchProduct(owner, productId);
```

Comment:
Let's not add the final keyword to [existing] code unnecessarily. I'm begrudgingly accepting of it in net-new code, but let's not go adding churn to pre-existing stuff for it.

Reply:
Ok. I can tune it a bit down. I like to be explicit with things like mutability. ... Why you don't like final?

Reply:
In Java, final on non-static, scoped variables does very little. It functions like \*const does in C -- we can't change the reference, but we can still change the object we're referencing. And due to Java's pass-by-value calling convention, the final operator does little more than add six characters to a given variable declaration. In the case of these methods, the value will only change if another developer comes in to change the logic; and if they need to change the value (as is common when working with the entities), the devs will either just remove the modifier, or redeclare it as non-final, wasting another 4 bytes of stack space.

In C/C++ and other such languages, it has a lot more value beyond just some syntax (even though you can cast it off), but in Java, it does almost nothing for us.